server/controllers/file.controller.js

```diff
@@ -38,8 +40,15 @@ function getAllDescendantIds(files, nodeId) {
 }
 
 function deleteMany(files, ids) {
-  ids.forEach(id => {
+  let s3ObjectUrlList = [];
+  each(ids, (id, callback) => {
+    if (files.id(id).url !== undefined) {
+      s3ObjectUrlList.push(files.id(id).url);
+    }
     files.id(id).remove();
+    callback();    
+  }, (err) => {
```

Comment:
3rd parameter (err) => {} is called when [async](https://caolan.github.io/async/docs.html#each) for all the element gets completed or hits an error. Thus, upon completion ie getting urlList I had given the list to deleteObjectsFromS3.